### ISSUE TITLE
Doesn't start after recent Synology DSM 6.2 updates

### ISSUE THREAD

**Issue by fastcat** [zerotier/ZeroTierNAS#41]
After installing the latest DSM 6.2 security update today (DSM 6.2.1-23824 Update 4 on DS1819+), ZT is no longer working.  The problem is that the `tun` module is no longer getting automatically loaded, and so ZT of course fails to start.

I'm not sure how other VPN packages handle this, but it seems to me that ZT's startup script needs to `insmod /lib/modules/tun.ko` if `/dev/net/tun` does not exist?  Note that, AFAICT, `modprobe` is wholly non-functional on the Synology :disappointed: 

Manually doing the `insmod` and then restarting the ZT service fixes things for me.

**Comment by fastcat** [comment]
It looks like actually the problem is that the current packages don't match the git repo ... they have done away with the watchdog wrapper script, which was responsible for inserting the `tun.ko` module.

That script is still present in this git repo ... but I don't know what happened to it in the packages at https://download.zerotier.com/RELEASES/1.2.12/dist/

**Comment by joseph-henry** [comment]
Thanks for identifying this issue. I've modified to script to helpfully address this. Can you give one of the packages here a try? (I believe your NAS would need the Denverton package)

http://download.zerotier.com/RELEASES/1.2.13/dist/synology/

**Comment by fastcat** [comment]
Well, regardless of whether it fixed this issue, upgrading the package wiped the configuration, so I consider that a definite fail.

**Comment by fastcat** [comment]
It does seem to have fixed the `tun.ko` loading issue, however...

Here's some stuff from `/var/log/packages/zerotier.log` from the upgrade that would seem to point out why the upgrade deleted my config:

```
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 Begin start-stop-status stop
11631
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 End start-stop-status stop ret=[0]
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 Begin preupgrade
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 End preupgrade ret=[0]
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 Begin preuninst
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 End preuninst ret=[0]
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 Begin /bin/rm -rf /volume1/@appstore/zerotier
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 End /bin/rm -rf /volume1/@appstore/zerotier ret=[0]
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 Begin postuninst
2019/06/10 10:47:47     upgrade zerotier 1.2.12-1 End postuninst ret=[0]
2019/06/10 10:47:47     upgrade zerotier 1.2.13-1 Begin preinst
2019/06/10 10:47:47     upgrade zerotier 1.2.13-1 End preinst ret=[0]
2019/06/10 10:47:47     upgrade zerotier 1.2.13-1 Begin /bin/mv -f /volume1/@tmp/pkginstall/package /volume1/@appstore/zerotier
2019/06/10 10:47:47     upgrade zerotier 1.2.13-1 End /bin/mv -f /volume1/@tmp/pkginstall/package /volume1/@appstore/zerotier ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin /bin/mkdir -p /var/packages/zerotier
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End /bin/mkdir -p /var/packages/zerotier ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin /bin/mv -f /volume1/@tmp/pkginstall/INFO /var/packages/zerotier/INFO
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End /bin/mv -f /volume1/@tmp/pkginstall/INFO /var/packages/zerotier/INFO ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin /bin/rm -rf /var/packages/zerotier/scripts
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End /bin/rm -rf /var/packages/zerotier/scripts ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin /bin/mv -f /volume1/@tmp/pkginstall/scripts /var/packages/zerotier/scripts
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End /bin/mv -f /volume1/@tmp/pkginstall/scripts /var/packages/zerotier/scripts ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin /bin/rm -rf /var/packages/zerotier/conf
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End /bin/rm -rf /var/packages/zerotier/conf ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin /bin/mv -f /volume1/@tmp/pkginstall/conf /var/packages/zerotier/conf
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End /bin/mv -f /volume1/@tmp/pkginstall/conf /var/packages/zerotier/conf ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin postinst
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End postinst ret=[0]
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 Begin postupgrade
2019/06/10 10:47:48     upgrade zerotier 1.2.13-1 End postupgrade ret=[0]
2019/06/10 10:47:49     upgrade zerotier 1.2.13-1 Begin start-stop-status start
insmod: ERROR: could not insert module /lib/modules/tun.ko: File exists
2019/06/10 10:47:50     upgrade zerotier 1.2.13-1 End start-stop-status start ret=[0]
2019/06/10 10:47:50     upgrade zerotier 1.2.13-1 Begin /bin/rm -rf /volume1/@tmp/pkginstall
2019/06/10 10:47:50     upgrade zerotier 1.2.13-1 End /bin/rm -rf /volume1/@tmp/pkginstall ret=[0]
2019/06/10 10:49:16     stop zerotier 1.2.13-1 Begin start-stop-status stop
2672
2019/06/10 10:49:16     stop zerotier 1.2.13-1 End start-stop-status stop ret=[0]
2019/06/10 10:50:29     start zerotier 1.2.13-1 Begin start-stop-status start
insmod: ERROR: could not insert module /lib/modules/tun.ko: File exists
2019/06/10 10:50:30     start zerotier 1.2.13-1 End start-stop-status start ret=[0]
2019/06/10 10:53:26     (system) trigger zerotier 1.2.13-1 Begin start-stop-status stop
4643
2019/06/10 10:53:26     (system) trigger zerotier 1.2.13-1 End start-stop-status stop ret=[0]
2019/06/10 10:55:00     (system) trigger zerotier 1.2.13-1 Begin start-stop-status start
2019/06/10 10:55:02     (system) trigger zerotier 1.2.13-1 End start-stop-status start ret=[0]
```

Note the `rm -rf /volume1/@appstore/zerotier` -- that's ... pretty fatal thing to be doing during an _upgrade_, since that's where all the config lives.

**Comment by joseph-henry** [comment]
Hey, thanks for your investigative footwork on this issue @fastcat. Here's an [updated set of packages](https://download.zerotier.com/PRERELEASES/1.4.0pre/synology) to preview `1.4.0`. It addresses the following issues:

 - Losing interface on reboot: `insmod /lib/modules/tun.ko` issued upon each reboot (and start of the service). See tickets #41 and #43 
 - Identity files moved to more traditional location `/var/lib/zerotier-one`, and are now preserved across uninstalls, reinstalls, updates, etc. Prior to this version, during uninstall we were intentionally deleting your identity files just for thoroughness but this seems to be annoying to most users (rightly so). So uninstalling and updating to this version is the last time that your configuration will get wiped. See tickets #41
 - Service script now sends proper SIGKILL signal
 - Segfault on ARM (*might* be fixed), see ticket #52
 - Memory leak (*might* be fixed), see ticket #46
